packages/blade/src/components/Box/BaseBox/useMemoizedStyles.web.ts

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import type { CSSObject } from 'styled-components';
+import { getBaseBoxStyles } from './getBaseBoxStyles';
+import type { BaseBoxProps } from './types';
+import type { Theme } from '~components/BladeProvider';
+
+const getDependencyProps = (props: BaseBoxProps & { theme?: Theme }): string | BaseBoxProps => {
```

Comment:
Optional, might be clearer to specify:

```suggestion
const getDependencyPropsSerialized = (props: BaseBoxProps & { theme?: Theme }): string | BaseBoxProps => {
```

Reply:
It may not be serialized if it gets returned from catch so kept it generic for now

Reply:
changed it to this - `getMemoDependency`. It inside memo's file so might help now?